File: sdks/php/src/client/src/Templates/Types/GenerateTemplatePreviewsRequestFormatOpts.php
<?php

namespace Pogodoc\Templates\Types;

use Pogodoc\Core\Json\JsonSerializableType;
use Pogodoc\Core\Json\JsonProperty;

/**
 * Format options for the rendered document
 */
class GenerateTemplatePreviewsRequestFormatOpts extends JsonSerializableType
{
    /**
     * @var ?float $fromPage
     */
    #[JsonProperty('fromPage')]
    public ?float $fromPage;

    /**
     * @var ?float $toPage
     */
    #[JsonProperty('toPage')]
    public ?float $toPage;

    /**
     * @var ?value-of<GenerateTemplatePreviewsRequestFormatOptsFormat> $format
     */
    #[JsonProperty('format')]
    public ?string $format;

    /**
     * @var ?string $waitForSelector Selector to wait for to know when the page is loaded and can be saved as pdf, png, etc.
     */
    #[JsonProperty('waitForSelector')]
    public ?string $waitForSelector;

    /**
     * @param array{
     *   fromPage?: ?float,
     *   toPage?: ?float,
     *   format?: ?value-of<GenerateTemplatePreviewsRequestFormatOptsFormat>,
     *   waitForSelector?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->fromPage = $values['fromPage'] ?? null;
        $this->toPage = $values['toPage'] ?? null;
        $this->format = $values['format'] ?? null;
        $this->waitForSelector = $values['waitForSelector'] ?? null;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/csharp/client/src/PogodocApi/Templates/Types/GenerateTemplatePreviewsRequestFormatOpts.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using PogodocApi.Core;

namespace PogodocApi;

/// <summary>
/// Format options for the rendered document
/// </summary>
[Serializable]
public record GenerateTemplatePreviewsRequestFormatOpts : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("fromPage")]
    public double? FromPage { get; set; }

    [JsonPropertyName("toPage")]
    public double? ToPage { get; set; }

    [JsonPropertyName("format")]
    public GenerateTemplatePreviewsRequestFormatOptsFormat? Format { get; set; }

    /// <summary>
    /// Selector to wait for to know when the page is loaded and can be saved as pdf, png, etc.
    /// </summary>
    [JsonPropertyName("waitForSelector")]
    public string? WaitForSelector { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
